mtls-enforcement/mtls-header-enforcement-mediator/src/main/java/org/wso2/financial/services/apim/mediation/policies/mtls/enforcement/MTLSHeaderEnforcementMediator.java

```diff
@@ -50,15 +56,16 @@ public boolean mediate(MessageContext messageContext) {
             Certificate certificate = MTLSEnforcementUtils
                     .getClientCertificateFromHeader(certificateHeaderValue, isClientCertificateEncoded);
             if (certificate == null) {
-                // TODO: handle error properly
-                return false;
+                String errorDescription = "Certificate not found in the header";
+                log.error(errorDescription);
+                setErrorResponseProperties(messageContext, "Unauthorized", errorDescription, "401");
+                throw new SynapseException(errorDescription);
             }
-        } catch (UnsupportedEncodingException e) {
-            // TODO: handle error properly
-            return false;
-        } catch (CertificateException e) {
-            // TODO: handle error properly
-            return false;
+        } catch (UnsupportedEncodingException | CertificateException e) {
+            String errorDescription = "Error parsing the certificate from the header";
+            log.error(errorDescription);
```

Comment:
shall we log the error e clause also

Reply:
Fixed with 02de70c4510b4383ba3f68084ee7b52b3f9bc688